src/wallet/wallet.cpp

```diff
@@ -3987,37 +3991,54 @@ CWallet* CWallet::CreateWalletFromFile(const std::string& name, const fs::path&
             return nullptr;
         }
         walletInstance->SetMaxVersion(nMaxVersion);
-    }
 
-    if (fFirstRun)
-    {
         // ensure this wallet.dat can only be opened by clients supporting HD with chain split and expects no default key
-        if (!gArgs.GetBoolArg("-usehd", true)) {
+        if (!gArgs.GetBoolArg("-usehd", true) && fFirstRun) {
             InitError(strprintf(_("Error creating %s: You can't create non-HD wallets with this version."), walletFile));
             return nullptr;
         }
-        walletInstance->SetMinVersion(FEATURE_NO_DEFAULT_KEY);
```

Comment:
Why is it ok to remove this if (fFirstRun) SetMinVersion(NO_DEFAULT_KEY)?

Reply:
`-upgradewallet` is implicitly true (and 0) on the first run. So the wallet version will be set to `FEATURE_LATEST`which is `FEATURE_NO_DEFAULT_KEY`.

Reply:
Hmm, but what if I override it? Can I start my node accidentally with upgradewallet=0 have it create a wallet and end up with a borked wallet?

Reply:
Still needs addressing, I think -upgradewallet=0 may break first-init after this patch.

Reply:
I changed the if statement to `if (gArgs.GetBoolArg("-upgradewallet", fFirstRun) || fFirstRun)`

Reply:
I still dont think this is right - I dont see where an fFirstRun wallet with upgradewallet=something non-0 in the bitcoin.conf will get its MinVersion set to at least FEATURE_NO_DEFAULT_KEY.